### PR TITLE
💳 Implement move founds modal on apply for opening

### PR DESCRIPTION
**Pull request by wyrdrender**
* Add new modal in case of insufficient balance
* Refactor move founds modal to meet useHasRequiredStake value
* WorkingGroupsOpenings eslint errors

### CODE REVIEW

File: packages/ui/src/working-groups/components/OpeningsList/OpeningDetails.tsx [hunk 1]
```diff
@@ -20,6 +23,36 @@ import { isOpeningOpen } from '@/working-groups/model/isOpeningOpen'
...
+          },
+        })
+      } else {
+        showModal<MoveFundsModalCall>({
+          modal: 'MoveFundsModal',
+          data: { type: MoveFoundsModalType.NO_FOUNDS, requiredStake },
+        })
+      }
+    }
+  }, [hasRequiredStake, transferableAccounts])
```
Comment (jodator) [inline packages/ui/src/working-groups/components/OpeningsList/OpeningDetails.tsx:55]: I would move that to the `ApplyForRoleModal` this way we don't have to copy that check when invoking some modal. I don't recall if we have only one place with this button but there's a chance that we have at least two.

Comment (jodator) [reply packages/ui/src/working-groups/components/OpeningsList/OpeningDetails.tsx:55]: The other thing is that the main modal, like ApplyForRole modal, could check only for `hasRequiredStake` and display the `<MoveFundsModal />`. The inner checks for which variant to display might be moved to the MoveFundsModall as well.

**Review by jodator — CHANGES_REQUESTED** [review]
Looks good :) But I think that the check should be done inside the ApplyForRole modal.

File: packages/ui/src/working-groups/components/OpeningsList/OpeningDetails.tsx (hunk 1 above)
Comment (wyrdrender) [reply packages/ui/src/working-groups/components/OpeningsList/OpeningDetails.tsx:55]: Sorry, there was a slight misunderstanding. I understood that you wanted to rid the modal of all logic. I'll fix it 👍

**Review by jodator — APPROVED** [review]

**Pull request merged**